DecipherNow/helm-charts
# Template spire ca

**Please provide enough information so that others can review your pull request:**

1. Explain the **details** for making this change. What existing problem does the pull request solve? If it resolves an existing issue, be sure to use a Github [keyword](https://help.github.com/en/articles/closing-issues-using-keywords) to automatically close it.

closes #661 

This PR adds make command `make spire-custom-ca` and script to automatically generate an upstream CA using acert and replace the necessary secrets files with these new values.

It also allows you to change the spire namespace by setting global.spire.namespace.

<br/><br/>

2. What **changes to custom.yaml** are required?

None 

<br/><br/>



To test:

Have acert installed https://github.com/deciphernow/acert

1. Run `make spire-custom-ca` and follow the prompts - enter the intended spire namespace, and then the args for acert authorities create.

    Update your global.spire.trust_domain and global.spire.namespace (if you changed it) as desired.

2. Run `make install`.

    Everything should come up and the mesh config secrets should use your configured trust domain.  Verify that the mesh 
    comes up correctly and you can access service endpoints through edge.

Exec into the spire server pod `kubectl exec -it server-0 -n spire -c server -- /bin/sh` and run ` /opt/spire/bin/spire-server entry show -registrationUDSPath /run/spire/socket/registration.sock`. Your entries should also have the new trust domain.

To see the certs being presented, pick a service and get its ip `kubectl describe pod {pod-id} | grep IP` and then exec into the data internal container and use openssl to look at the cert: `kubectl exec -it data-internal-0 -c data-internal -- /bin/sh` and run `openssh s_client --connect {IP}:10808`.
